src/app/login/login-home/login-home.component.spec.ts

```diff
@@ -1,74 +1,139 @@
-    };
-    return Observable.create((observer) => {
-      observer.next(config);
-      observer.complete();
-    });
-  }
-  getContextPath(): string {
-    return '/wise';
-  }
-  getRecaptchaPublicKey(): string {
-    return '';
-  }
-}
+import { RECAPTCHA_V3_SITE_KEY, ReCaptchaV3Service, RecaptchaV3Module } from 'ng-recaptcha';
+import { By } from '@angular/platform-browser';
+import { HttpClient } from '@angular/common/http';
+import { Router } from '@angular/router';
+import { getErrorMessage } from '../../common/test-helper';
 
 let component: LoginHomeComponent;
+let configService: ConfigService;
+const contextPath: string = '/wise';
+let fixture: ComponentFixture<LoginHomeComponent>;
+let http: HttpClient;
+const recaptchaPrivateKey: string = 'the-private-key';
+let recaptchaV3Service: ReCaptchaV3Service;
+const redirectUrl: string = `${contextPath}/api/j_acegi_security_check`;
+let router: Router;
+let userService: UserService;
+
 describe('LoginHomeComponent', () => {
-  let fixture: ComponentFixture<LoginHomeComponent>;
   beforeEach(
     waitForAsync(() => {
       TestBed.configureTestingModule({
         declarations: [LoginHomeComponent],
-        imports: [HttpClientTestingModule, RouterTestingModule, RecaptchaModule],
+        imports: [HttpClientTestingModule, RecaptchaV3Module, RouterTestingModule],
         providers: [
-          { provide: UserService, useClass: MockUserService },
-          { provide: ConfigService, useClass: MockConfigService }
+          ConfigService,
+          { provide: RECAPTCHA_V3_SITE_KEY, useValue: recaptchaPrivateKey },
+          ReCaptchaV3Service,
+          UserService
         ],
         schemas: [NO_ERRORS_SCHEMA]
       }).compileComponents();
     })
   );
 
   beforeEach(() => {
+    configService = TestBed.inject(ConfigService);
+    http = TestBed.inject(HttpClient);
+    recaptchaV3Service = TestBed.inject(ReCaptchaV3Service);
+    router = TestBed.inject(Router);
+    userService = TestBed.inject(UserService);
+    spyOn(userService, 'getRedirectUrl').and.returnValue(redirectUrl);
+    spyOn(configService, 'getRecaptchaPublicKey').and.returnValue('the-public-key');
+    spyOn(configService, 'getContextPath').and.returnValue(contextPath);
     fixture = TestBed.createComponent(LoginHomeComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
+
   getRedirectUrl();
+  login();
 });
 
 function getRedirectUrl() {
   describe('getRedirectUrl()', () => {
     it('should add redirectUrl to Google login url', () => {
       expect(component.getRedirectUrl('google')).toEqual(
-        '/wise/api/google-login?redirectUrl=/path/to/redirect'
+        `${contextPath}/api/google-login?redirectUrl=${redirectUrl}`
       );
     });
   });
 }
+
+function login() {
+  describe('login()', () => {
+    loginWithRecaptchaDisabled();
+    loginWithRecaptchaEnabled();
+  });
+}
+
+function loginWithRecaptchaDisabled() {
```

Comment:
Code Climate flagged this as a long function (32 lines, exceeds 25 allowed). Break up into sub-functions?